make.js

```diff
@@ -687,3 +688,47 @@ target.gensprintlyzip = function() {
 
     console.log('\n# Completed creating sprintly zip.');
 }
+
+// Verify that no tasks have min agent demands greater than whatever version of the agent is fully rolled out.
+target.verifyMinAgentDemands = function() {
+    console.log("Verifying min agent demands.");
+    var octokit = new Octokit();
+    octokit.repos.getLatestRelease({
+        owner: "microsoft", 
+        repo: "azure-pipelines-agent"
+    })
+    .then(({data}) => {
+        // Find the version of the agent that is fully rolled out
+        var agentVersion = data.name.substr(1);
+        console.log(`Latest version of the Agent that's fully rolled out is ${agentVersion}.`);
+
+        // Iterate all tasks and make sure none of them depend on a version higher than what's rolled out
+        var invalidMinAgentDemandsExist = false;
+        taskList.forEach(function(taskName) {
+            var taskPath = path.join(__dirname, 'Tasks', taskName);
+            ensureExists(taskPath);
+    
+            // Load the task.json
+            var taskJsonPath = path.join(taskPath, 'task.json');
+            if (test('-f', taskJsonPath)) {
+                var taskDef = fileToJson(taskJsonPath);
+                
+                if (taskDef.minimumAgentVersion)
+                {
+                    if (semver.gt(taskDef.minimumAgentVersion, agentVersion)) {
+                        console.log(`Error! Task ${taskName} has a minimum agent version of ${taskDef.minimumAgentVersion} but the latest version of the Agent is ${agentVersion}.`);
```

Comment:
This should probably be a task-lib.error so we get nice output

Reply:
I will keep this and add `tl.error` it where we `exit(1)`. Wanted to keep console log in case > 1 tasks fail the user can check logs and see version issues.

Reply:
JK. That's probably rare.

Reply:
How about we use logissue - https://github.com/microsoft/azure-pipelines-tasks/blob/master/docs/authoring/commands.md#task-logging-commands